### DISCUSSION TITLE
Update a new element if some condition is met

### DISCUSSION THREAD

**Discussion by sgandon** [mikefarah/yq#617] [Q&A] [answered]
Hello and happy to be the first to use this discussion feature.
I am using a for loop to update many yaml files and I'd like to find a yq expression to only update the metadata.name value if the kind == "Namespace" and not update the file if not.
here the yaml file I'd like to update.
```
apiVersion: v1
kind: Namespace
metadata:
  name: foo
```
Is it possible with yq 4 ?

**Comment by mikefarah** [comment, marked as answer]
I've got:

```
yq e '(select(.kind == "Namespace") | .metadata.name) = "meow"' sample.yaml
```

**Reply by mikefarah** [reply]
Also - the title says 'add' but the description is 'update'? I'm assuming the description is right (could you update the title?)

**Reply by dlampa** [reply]
Thanks a lot, this has helped me a lot. Could you perhaps provide an explanation of the expression syntax, i.e. what is the significance of having ( ) around on the LHS of the assignment when referring to the example:
```
foo: bar
bar: baz
items:
  - item:
      name: foo
  - item:
      name: abc
  - item:
      name: def
something else: foobar
```
`yq e '(.items.[] | select(.item.name=="foo") | .item.address) ="test" ' sample.yml`
vs.
`yq e '.items.[] | select(.item.name=="foo") | .item.address="test" ' sample.yml`

Thanks!!

**Comment by sgandon** [comment]
Thanks a lot for your answer, and I'll change the title. 
In fact I have the 3 following requirements : update, add and remove one attribute base on a selected node. But from your suggestion, I figured the rest.
